### ISSUE TITLE
Editor crashes after creating a camera entity without a loaded level

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#4463]
**Describe the bug**
Editor crashes when a level is not loaded and a camera entity is created through the viewport context menu.

**To Reproduce**
1. Launch the Editor.
2. Close the "Welcome to O3DE" screen.
3. Press RMB in the viewport and select "Create camera entity from view".

**Expected behavior**
It is not possible to create a camera entity without a loaded level. Viewport context menu is disabled until a level is loaded.

**Call Stack Trace:**
```
25) 00007FFA6076C347 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFA62C951B0 (ntdll) : memset
23) 00007FFA62C7C766 (ntdll) : _C_specific_handler
22) 00007FFA62C920CF (ntdll) : _chkstk
21) 00007FFA62C41454 (ntdll) : RtlRaiseException
20) 00007FFA62C90BFE (ntdll) : KiUserExceptionDispatcher
19) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\Template\Template.cpp (69) : AzToolsFramework::Prefab::Template::IsValid
18) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabSystemComponent.cpp (387) : AzToolsFramework::Prefab::PrefabSystemComponent::FindTemplateDom
17) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\Instance\InstanceToTemplatePropagator.cpp (161) : AzToolsFramework::Prefab::InstanceToTemplatePropagator::PatchTemplate
16) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabUndoHelpers.cpp (30) : AzToolsFramework::Prefab::PrefabUndoHelpers::UpdatePrefabInstance
15) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Entity\PrefabEditorEntityOwnershipService.cpp (116) : AzToolsFramework::PrefabEditorEntityOwnershipService::AddEntity
14) C:\O3DE\o3de\Code\Framework\AzFramework\AzFramework\Entity\EntityContext.cpp (229) : AzFramework::EntityContext::CreateEntity
13) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Entity\EditorEntityContextComponent.cpp (229) : AzToolsFramework::EditorEntityContextComponent::CreateNewEditorEntity
12) C:\O3DE\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1556) : AZ::Internal::EBusContainer<AzToolsFramework::EditorEntityContextRequests,AzToolsFramework::EditorEntityContextRequests,0,0>::Dispatcher<AZ::EBus<AzToolsFramework::EditorEnti
11) C:\O3DE\o3de\Gems\Camera\Code\Source\CameraEditorSystemComponent.cpp (97) : Camera::CameraEditorSystemComponent::CreateCameraEntityFromViewport
10) 00007FFA12F80DB9 (Qt5Core) : QObject::qt_static_metacall
 9) 00007FFA1A6B5DCF (Qt5Widgets) : QAction::activate
 8) 00007FFA1A80C698 (Qt5Widgets) : QMenu::actionGeometry
 7) 00007FFA1A80C473 (Qt5Widgets) : QMenu::actionGeometry
 6) 00007FFA1A811C51 (Qt5Widgets) : QMenu::mouseReleaseEvent
 5) 00007FFA1A6E7B02 (Qt5Widgets) : QWidget::event
 4) 00007FFA1A80DC37 (Qt5Widgets) : QMenu::event
 3) 00007FFA1A6C49B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
 2) 00007FFA1A6C28D0 (Qt5Widgets) : QApplication::notify
 1) 00007FFA12F61B4A (Qt5Core) : QCoreApplication::notifyInternal2
```

**Screenshots**

https://user-images.githubusercontent.com/86952009/135848117-91441dd3-4696-4c42-aa55-7bb4da55a8df.mp4

**Desktop/Device:**

- Device: PC
- OS: Windows
- Version 10
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/7277859/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7277860/error.log)
[dump.zip](https://github.com/o3de/o3de/files/7277861/dump.zip)

**Comment by hultonha** [comment]
Assigned to `feature/prefab` @monroegm - this looks more related to Prefabs than camera to me. Let me know if you disagree. Thanks!

**Comment by LB-BartoszDudzinski** [comment]
Reopening as the issue still occurs in development branch (cdf5a05). The issue is fixed in Stabilization branch (d598a7c)).

**Comment by LB-BartoszDudzinski** [comment]
Verified as Fixed in Development.
